File: 2171-removing-minimum-number-of-magic-beans/2171-removing-minimum-number-of-magic-beans.py
class Solution:
    def minimumRemoval(self, beans: List[int]) -> int:
        min_cnt = float('inf')
        beans.sort(reverse = True)
        summ = sum(beans)
        while beans:
            minn = beans[-1]
            cnt = summ - minn * len(beans)
            min_cnt = min(min_cnt, cnt)
            beans.pop()
                
        return min_cnt

File: 2171-removing-minimum-number-of-magic-beans/2171-removing-minimum-number-of-magic-beans.kt
import kotlin.math.*

class Solution {
    fun minimumRemoval(beans: IntArray): Long {
        
        beans.sort()
        var n = beans.size
        var summ : Long = 0L
        for (bean in beans){
            summ += bean.toLong()
        }
        var minn = Long.MAX_VALUE
        
        for (i in 0 until n){
            val outs = summ - (beans[i].toLong() * (n - i).toLong())
            if (outs >= 0)
                minn = min(outs, minn)
        }
        return minn
    }
}